### PR TITLE
fix: Change list url of Event and Ticket Discount Code. 

### LINKED ISSUE #4937: Separate Discount code into EventDiscountCode and TicketDiscountCode
Right now the Discount code schema/model/endpoints they handle two types of Discount codes. 
1. The Event Discount code created by the admin and applied on the service charges.
2. The Ticket discount code created by the event organizer and applied on the ticket prices of that event. 

This is IMO not following separation of concerns and a source of a lot of bugs. It is extremely confusing to the developers specially the new ones. It is well explained in the dev handbook but I am not sure how many developers would go through the dev-handbook before contributing to the project. For the client side devs of frontend and android, it is even worse. 

Consider the two endpoints:
1. `/v1/events/{event_identifier}/discount-codes` - List of ticket discount codes under an event.
2. `/v1/events/{event_identifier}/discount-code` - Event discount code related with that event.

The difference between the two endpoints is a bare `s`. It seems highly confusing to me atleast if not others. 
I propose to separate them out into two endpoints. If there are lot of common fields between them we can always create a base class and let the two classes extend them in order to obey DRY. 
I am also not sure about the priority of this issue as to whether we should do it after successful feature parity with v1 and release or before that but I think we should definitely fix this one time or the other. 

Thoughts: @iamareebjamal @niranjan94 @SaptakS @poush @mayank8318 @schedutron @bhaveshAn

### PR DESCRIPTION
**Pull request by prateekj117**
Fixes #4937 

#### Checklist

- [X] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [X] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)

- [X] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

File: app/api/__init__.py [hunk 1]
```diff
@@ -469,11 +469,11 @@
 
 # discount codes
 api.route(DiscountCodeListPost, 'discount_code_list_post', '/discount-codes')
-api.route(DiscountCodeList, 'discount_code_list', '/events/<int:event_id>/discount-codes',
+api.route(DiscountCodeList, 'discount_code_list', '/events/<int:event_id>/ticket-discount-codes',
           '/events/<event_identifier>/discount-codes', '/users/<int:user_id>/discount-codes',
           '/tickets/<int:ticket_id>/discount-codes')
 api.route(DiscountCodeDetail, 'discount_code_detail', '/discount-codes/<int:id>',
-          '/events/<int:event_id>/discount-code', '/event-invoices/<int:event_invoice_id>/discount-code',
+          '/events/<int:event_id>/event-discount-codes', '/event-invoices/<int:event_invoice_id>/discount-code',
```
Comment (iamareebjamal) [inline app/api/__init__.py]: Why does DiscountCodeDetail have a URL of `/events/<int:event_id>/event-discount-codes`

Comment (prateekj117) [reply app/api/__init__.py]: @iamareebjamal I changed the appropriate Url names which were specifically for `event-discount-codes` and `ticket-discount-codes`.

Comment (iamareebjamal) [reply app/api/__init__.py]: But what will `/events/<int:event_id>/event-discount-codes` show. It is DiscountCodeDetail, not a list. Show me the response

Comment (prateekj117) [reply app/api/__init__.py]: @iamareebjamal Ok, I am confused. See [this](https://api.eventyay.com/#discount-codes-ticket-discount-code-collection-get) and [this](https://api.eventyay.com/#discount-codes-get-discount-code-detail-of-an-event-get), we are using the same Url to retrieve a list and to retrieve Detail of a discount code.

Comment (prateekj117) [reply app/api/__init__.py]: @iamareebjamal Yaa, it wouldn't give a list. It gives the detail of a discount code under an event, and not the discount code list which is of the type `event`. I will make appropriate changes. I got confused.

File: app/api/__init__.py [hunk 2]
```diff
@@ -469,7 +469,7 @@
 
 # discount codes
 api.route(DiscountCodeListPost, 'discount_code_list_post', '/discount-codes')
-api.route(DiscountCodeList, 'discount_code_list', '/events/<int:event_id>/discount-codes',
+api.route(DiscountCodeList, 'discount_code_list', '/events/<int:event_id>/ticket-discount-codes',
```
Comment (shreyanshdwivedi) [inline app/api/__init__.py]: You are using `ticket-discount-codes` and `event-discount-codes` in docs and test but I don't see `event-discount-codes` here? Also, did you open an issue to accompany these changes on FE?

Comment (prateekj117) [reply app/api/__init__.py]: @shreyanshdwivedi I got confused. The Url to create ticket discount code is also wrong here. I will open a separate PR for that. And correct the changes here.

Comment (prateekj117) [reply app/api/__init__.py]: @shreyanshdwivedi Done. Please review now.

**Comment by prateekj117** [comment]
@CosmicCoder96 The issue was not about a separate endpoint, but about naming the endpoint more appropriately.

**Comment by CosmicCoder96** [comment]
If the name of the model  is `discount-code`, and the url is `/discount-codes`, then why does it need to be renamed, if separate endpoints are serving no purpose?
You are indeed creating two new endpoints for discount codes in this PR.

**Comment by prateekj117** [comment]
@CosmicCoder96 The issue description gives an example of how a difference of basic 's' gives two separate results. I just renamed one of the endpoints to something more appropriate. Also, the Url of creating ticket discount code ([this](https://api.eventyay.com/#discount-codes-ticket-discount-code-collection-post)) is wrong. Opening a separate issue for that.

**Comment by CosmicCoder96** [comment]
Yes, but at the expense of making manual GET requests, to get the same thing. FE can handle this easily (already does via properly defined relationships), and if for android it doesn't matter if it is following the JSON API spec or not, then we don't need to break away from the JSON API convention, this issue is not valid in my opinion. @iamareebjamal Do we really need this?

**Comment by prateekj117** [comment]
@CosmicCoder96 Ya, agree on this.

**Comment by iamareebjamal** [comment]
@CosmicCoder96 No, that's why I commented on the issue "low priority'

**Comment by prateekj117** [comment]
@CosmicCoder96 @iamareebjamal Solved it because it was discussed in the weekly meeting and was told to be solved within last week. Sorry for wasting your time. Closing this.

**Pull request closed without merging**